plutus-core/plutus-ir/src/PlutusIR/Core/Instance/Pretty/Readable.hs

```diff
@@ -0,0 +1,196 @@
+{-# LANGUAGE ConstraintKinds       #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE UndecidableInstances  #-}
+{-# LANGUAGE ViewPatterns          #-}
+{-# OPTIONS_GHC -Wno-orphans       #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+{-# OPTIONS_GHC -Wno-incomplete-uni-patterns #-}
+module PlutusIR.Core.Instance.Pretty.Readable
+  ( prettyPirReadable
+  ) where
+
+import PlutusCore.Default.Universe
+import PlutusCore.Pretty.ConfigName
+import PlutusCore.Pretty.PrettyConst
+import PlutusCore.Pretty.Readable
+import PlutusIR.Core.Type
+import PlutusPrelude
+import Prettyprinter
+import Prettyprinter.Custom
+
+import Data.Default.Class
+
+type PrettyPir = PrettyBy (PrettyConfigReadable PrettyConfigName)
+
+-- | Pretty-print something with the Pir prettyprinter settings.
+prettyPirReadable :: PrettyPir a => a -> Doc ann
+prettyPirReadable = prettyBy prettyConfigReadable
+  -- Note, for now this is just a simple default setting. We may want to add more complicated
+  -- settings in the future.
+  where prettyConfigReadable = botPrettyConfigReadable defPrettyConfigName def
+
+-- | Split an application into its (possible) head and arguments (either types or term)
+viewApp :: Term tyname name uni fun ann
```

Comment:
TODO: put these in a utility module somewhere, they're generically useful.

Reply:
I'll let you decide where to put them.